src/coreclr/jit/importercalls.cpp

```diff
@@ -2759,6 +2759,32 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Runtime_CompilerServices_RuntimeHelpers_IsReference:
+            {
+                assert(sig->sigInst.methInstCount == 1);
+                CORINFO_CLASS_HANDLE hClass = sig->sigInst.methInst[0];
+                retNode                     = gtNewIconNode(eeIsValueClass(hClass) ? 0 : 1);
+                break;
+            }
+
+            case NI_System_Runtime_CompilerServices_RuntimeHelpers_IsReferenceOrContainsReferences:
+            {
+                assert(sig->sigInst.methInstCount == 1);
+                CORINFO_CLASS_HANDLE hClass = sig->sigInst.methInst[0];
+                retNode =
+                    gtNewIconNode(((info.compCompHnd->getClassAttribs(hClass) &
+                                    (CORINFO_FLG_VALUECLASS | CORINFO_FLG_CONTAINS_GC_PTR)) == CORINFO_FLG_VALUECLASS)
```

Comment:
same here, it seems like it's not exactly the same check as was on VM side?

Reply:
dtto